ci/docker/build.Dockerfile

```diff
@@ -9,9 +9,6 @@ ENV BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
 ENV BUILDKITE_COMMIT=${BUILDKITE_COMMIT}
 ENV BUILDKITE_PULL_REQUEST_BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
 ENV TRAVIS_COMMIT=${BUILDKITE_COMMIT}
-# Set compiler here to build Ray with CLANG/LLVM
-ENV CC=clang
-ENV CXX=clang++-12
```

Comment:
why this is required?

Reply:
I think these flags somehow mess up the build. https://buildkite.com/ray-project/oss-ci-build-pr/builds/32272#018a0079-0255-4c9d-bb9d-095edf048e65

I'm not expert there and it took me a long time to realize it's because of these flags.

Another thing is that we build linux wheel with gcc (manylinux docker), so it seems doesn't make sense using clang to run CI.

It's added by this PR, https://github.com/ray-project/ray/pull/31522 

I'll see whether it can pass the arm build or not by just removing this. Otherwise, I'll add a flag there to enable it only for arm build.

Reply:
hmm.. if you want to make this change, could you put it in a separate PR? I guess the change is not tightly tied to this change in this PR, right?

Reply:
sure. stil working on this one.